From bcec98c5afa7665202007d52bc1b79d70243a23b Mon Sep 17 00:00:00 2001
From: Bhakthavatsala Raghavendra <bhaktha@google.com>
Date: Thu, 12 Mar 2026 15:03:14 -0700
Subject: [PATCH 5/5] Avoid buffer overflow

Bug: 484861632
Flag: EXEMPT BUGFIX
Test: m packages/module/Bluetooth
Change-Id: I29a981b57bca25a65575e29c788fdca13d5494eb
(cherry picked from commit a06794634568ac246276971376af4adff8e8d893)
---
 system/stack/l2cap/l2c_fcr.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/system/stack/l2cap/l2c_fcr.cc b/system/stack/l2cap/l2c_fcr.cc
index cc0bc0569a..40dbaa2a49 100644
--- a/system/stack/l2cap/l2c_fcr.cc
+++ b/system/stack/l2cap/l2c_fcr.cc
@@ -229,7 +229,7 @@ BT_HDR* l2c_fcr_clone_buf(BT_HDR* p_buf, uint16_t new_offset, uint16_t no_of_byt
    * NOTE: We allocate extra L2CAP_FCS_LEN octets, in case we need to put
    * the FCS (Frame Check Sequence) at the end of the buffer.
    */
-  uint16_t buf_size = no_of_bytes + sizeof(BT_HDR) + new_offset + L2CAP_FCS_LEN;
+  uint32_t buf_size = no_of_bytes + sizeof(BT_HDR) + new_offset + L2CAP_FCS_LEN;
   BT_HDR* p_buf2 = (BT_HDR*)osi_malloc(buf_size);
 
   p_buf2->offset = new_offset;
-- 
2.53.0

